Hi Sanjay, 

I have completed all the alignment of our RNA-seq data to Staphylococcus aureus (NC_007795)  and human genome (hg19). 

As the following showed, 90% reads are not located in human nor Staphylococcus aureus. Did you notice this result before? Or do you think I have some mistake in the analysis? 

Thanks. 

Shicheng


## Staphylococcus aureus (NC_007795)  
```
16_S3_L001.e568851:0.11% overall alignment rate
17_S4_L001.e568852:0.02% overall alignment rate
18_S1_L001.e568853:0.80% overall alignment rate
19_S2_L001.e568854:0.21% overall alignment rate
21_S3_L001.e568855:0.02% overall alignment rate
23_S3_L001.e568856:0.84% overall alignment rate
24_S2_L001.e568857:1.63% overall alignment rate
27_S1_L001.e568858:0.18% overall alignment rate
28_S4_L001.e568859:0.57% overall alignment rate
30_S3_L001.e568860:0.01% overall alignment rate
34_S1_L001.e568861:0.01% overall alignment rate
35_S4_L001.e568862:0.00% overall alignment rate
36_S1_L001.e568863:0.66% overall alignment rate
38_S2_L001.e568864:0.06% overall alignment rate
40_S2_L001.e568865:0.01% overall alignment rate
42_S3_L001.e568866:0.01% overall alignment rate
44_S3_L001.e568867:0.00% overall alignment rate
45_S1_L001.e568868:0.23% overall alignment rate
46_S4_L001.e568869:0.00% overall alignment rate
47_S4_L001.e568870:0.06% overall alignment rate
51_S2_L001.e568871:1.72% overall alignment rate
54_S4_L001.e568872:0.02% overall alignment rate
5_S1_L001.e568873:0.31% overall alignment rate
8_S2_L001.e568874:0.01% overall alignment rate
```
## human 
```
16_S3_L001Log.final.out:Uniquely mapped reads % |4.76%
17_S4_L001Log.final.out:Uniquely mapped reads % |0.79%
18_S1_L001Log.final.out:Uniquely mapped reads % |2.44%
19_S2_L001Log.final.out:Uniquely mapped reads % |1.75%
21_S3_L001Log.final.out:Uniquely mapped reads % |2.88%
23_S3_L001Log.final.out:Uniquely mapped reads % |4.86%
24_S2_L001Log.final.out:Uniquely mapped reads % |5.70%
27_S1_L001Log.final.out:Uniquely mapped reads % |31.47%
28_S4_L001Log.final.out:Uniquely mapped reads % |7.63%
30_S3_L001Log.final.out:Uniquely mapped reads % |0.07%
34_S1_L001Log.final.out:Uniquely mapped reads % |0.05%
35_S4_L001Log.final.out:Uniquely mapped reads % |0.05%
36_S1_L001Log.final.out:Uniquely mapped reads % |1.55%
38_S2_L001Log.final.out:Uniquely mapped reads % |0.76%
40_S2_L001Log.final.out:Uniquely mapped reads % |0.12%
42_S3_L001Log.final.out:Uniquely mapped reads % |0.05%
44_S3_L001Log.final.out:Uniquely mapped reads % |0.01%
45_S1_L001Log.final.out:Uniquely mapped reads % |6.82%
46_S4_L001Log.final.out:Uniquely mapped reads % |0.14%
47_S4_L001Log.final.out:Uniquely mapped reads % |0.11%
51_S2_L001Log.final.out:Uniquely mapped reads % |7.58%
54_S4_L001Log.final.out:Uniquely mapped reads % |5.68%
5_S1_L001Log.final.out:Uniquely mapped reads % |0.23%
8_S2_L001Log.final.out:Uniquely mapped reads % |0.14%
```
